packages/components/src/dropdown-menu/index.native.js

```diff
@@ -1,5 +1,173 @@
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import Dropdown from '../dropdown';
+
+function mergeProps( defaultProps = {}, props = {} ) {
+	const mergedProps = {
+		...defaultProps,
+		...props,
+	};
+
+	if ( props.className && defaultProps.className ) {
+		mergedProps.className = classnames(
+			props.className,
+			defaultProps.className
+		);
+	}
+
+	return mergedProps;
+}
+
+function DropdownMenu( {
+	children,
+	className,
+	controls,
+	icon = 'menu',
+	label,
+	popoverProps,
+	toggleProps,
+	// The following props exist for backward compatibility.
+	menuLabel,
+	position,
+} ) {
+	if ( menuLabel ) {
+		deprecated( '`menuLabel` prop in `DropdownComponent`', {
+			alternative: '`menuProps` object and its `aria-label` property',
+			plugin: 'Gutenberg',
+		} );
+	}
+
+	if ( position ) {
+		deprecated( '`position` prop in `DropdownComponent`', {
+			alternative: '`popoverProps` object and its `position` property',
+			plugin: 'Gutenberg',
+		} );
+	}
+
+	if ( isEmpty( controls ) && ! isFunction( children ) ) {
+		return null;
+	}
+
+	// Normalize controls to nested array of objects (sets of controls)
+	let controlSets;
+	if ( ! isEmpty( controls ) ) {
+		controlSets = controls;
+		if ( ! Array.isArray( controlSets[ 0 ] ) ) {
+			controlSets = [ controlSets ];
+		}
+	}
+	const mergedPopoverProps = mergeProps(
+		{
+			className: 'components-dropdown-menu__popover',
+			position,
+		},
+		popoverProps
+	);
+
+	return (
+		<Dropdown
+			className={ classnames( 'components-dropdown-menu', className ) }
+			popoverProps={ mergedPopoverProps }
+			renderToggle={ ( { isOpen, onToggle } ) => {
+				const openOnArrowDown = ( event ) => {
+					if ( ! isOpen && event.keyCode === DOWN ) {
+						event.preventDefault();
+						event.stopPropagation();
+						onToggle();
+					}
+				};
+				const mergedToggleProps = mergeProps(
+					{
+						className: classnames(
+							'components-dropdown-menu__toggle',
+							{
+								'is-opened': isOpen,
+							}
+						),
+					},
+					toggleProps
+				);
+
+				return (
+					<Button
+						{ ...mergedToggleProps }
+						icon={ icon }
+						onClick={ ( event ) => {
+							onToggle( event );
+							if ( mergedToggleProps.onClick ) {
+								mergedToggleProps.onClick( event );
+							}
+						} }
+						onKeyDown={ ( event ) => {
+							openOnArrowDown( event );
+							if ( mergedToggleProps.onKeyDown ) {
+								mergedToggleProps.onKeyDown( event );
+							}
+						} }
+						aria-haspopup="true"
+						aria-expanded={ isOpen }
+						label={ label }
+						showTooltip
+					>
+						{ mergedToggleProps.children }
+					</Button>
+				);
+			} }
+			renderContent={ ( { isOpen, onClose, ...props } ) => {
+				return (
+					<BottomSheet
+						hideHeader={ true }
+						isVisible={ isOpen }
+						onClose={ onClose }
+					>
+						{ isFunction( children ) ? children( props ) : null }
+						<PanelBody title={ label }>
```

Comment:
Could we use the Picker component to reuse functionality? (packages/components/src/mobile/picker/)

Reply:
The Picker is using an ActionSheet in iOS that doesn't translate to the interface we want here.
Or are you saying to refactor the Picker component to be like like this visually?

Reply:
Oh, I thought we wanted the ActionSheet on iOS based on @iamthomasbishop 's [comment](https://github.com/WordPress/gutenberg/pull/20745#issuecomment-599664171).

Still, as this layout is already in place somewhere else, maybe we could add a prop to the picker component, something like `useNativeActionSheet`, and use the code currently from the Android implementation if that prop is `false`. Anyway, not sure if this is the best way to go or if it will make things unnecessary complicated.